### PR TITLE
Dataproc: Addition of a dataproc cluster creation

### PR DESCRIPTION
**Pull request by rasalt**
Patchset includes changes to optionally include the creation
of a dataproc cluster

Signed-off-by: Ruchika k <rkharwar@google.com>

### CODE REVIEW

File: deploy/samples/project_with_local_audit_logs_dataproc.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
+# Copyright 2018 Google LLC
```
Comment (dufton) [inline deploy/samples/project_with_local_audit_logs_dataproc.yaml]: Maybe combine these two samples, and update the comment at the top of the file to describe what is deployed

File: deploy/project_config.yaml.schema [hunk 2]
```diff
@@ -250,6 +250,45 @@ definitions:
             minimum: 10
             maximum: 600
 
+      dataproc:
+        type: array
+        description: List of dataproc clusters to create.
+        items:
+          type: object
+          additionalProperties: true
```
Comment (dufton) [inline deploy/project_config.yaml.schema]: Does this block need additionalProperties? If not, it is useful to set it to false to catch typos in configs.

File: deploy/project_config.yaml.schema [hunk 3]
```diff
@@ -250,6 +250,45 @@ definitions:
...
+          required:
+          - name
+          - zone
+          - machine_type
+          - region
+          - workernum
```
Comment (dufton) [inline deploy/project_config.yaml.schema]: minor comment: num_workers, num_masters

**Comment by nikklassen** [comment]
This PR is now very out of date. Closing due to inactivity.

**Pull request closed without merging**